app-agent/src/exec_process.rs

```diff
@@ -17,7 +17,7 @@ use alumet::{
     plugin::event::StartConsumerMeasurement,
     resources::ResourceConsumer,
 };
-use anyhow::{anyhow, Context};
+use anyhow::{anyhow, Context, Error};
 
 /// Spawns a child process and waits for it to exit.
 pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result<ExitStatus> {
```

Comment:
Please include the error `e` in the returned errors, so that we don't lose the original information.